README.md
# Mint-Yz: The New Colors of a New Era (GTK theme)(Flatpak development fork)
**This repo exists solely to package Mint-Yz for Flathub. All contributions should be directed to the upstream repo https://github.com/SebastJava/mint-yz-theme, unless they are Flatpak/Flathub-specifc. This repo will attempt to stay near-even with upstream.**

Don't miss the **Mint-Yz-icons** with the same color scheme: https://github.com/SebastJava/mint-yz-icons

**The interface of all your software and the entire desktop is now rejuvenated, modernized. This is a fork from your reliable Mint-Y theme. It is the standard Linux Mint GTK theme now in a new edition with more energized and readable colors.**

That’s 11 different colors in 3 different variants: light, darker and darkest. This makes 33 different themes, all included in one little Debian package. And all of this is done on a solid foundation of improved shell and Python programs. It is also renamed **Mint-Yz** and is a separate Debian package. This means you can install it in addition to the standard **Mint-Y,** compare, experiment, and change your mind at any time. Your good old Mint-Y will not be removed or changed. And all of this is made very easy, see the installation instructions below.

![preview of the new desktop](Mint-Y-Colors/preview-nemo/preview-nemo-MintSoda.png)

![image of the new colors](Mint-Y-Colors/Mint-Y-Variations.png)

### Installation
First, check your system is not too old for this theme to work properly. This theme works well on Linux Mint 19, Linux Mint 20, and newer versions. (GTK+ 3.22 or newer)

1. Go to the [Releases](https://github.com/SebastJava/mint-yz-theme/releases) page. Select the latest.
1. Click on **Assets** to see the files if needed. (under the short presentation)
1. Download the **mint-yz-theme_x.x_all.deb** file. (x.x = version number)
1. Open it and click the **[Install Package]** button.
1. Select your new themes in **Menu > Preferences > Themes**

### Carefully chosen colors and precise engineering
This color branch here is the sum of a lot of work. At the start there was the Horst3180 Arc-theme. On this well-designed and solid foundation was created the Mint-Y theme, which has evolved considerably since. Then I created a fork of this Mint-Y and made some improvements on the auto branch, to get the build process 100% automated. And finally, this colors branch was created on top of all that.

The colors convey energies, vibrations... They should not be too subdued. But they should not always be set at full saturation and brightness either, because this could cause eye strain. So the saturation and lightness levels are carefully chosen here. And there is also the contrast with the white text in the foreground which received sustained attention. These new colors are therefore energizing, stimulating, and not tiring.

### Other design improvements
Version 2.0 and up comes with other design improvements:

![Other-improvements-preview](Other-improvements-preview/Other-improvements-preview.png)

  * Buttons and entries rounded corners are a bit more rounded, 3px->5px. I never liked this 3px border-radius. This makes button’s rounded corners so subtle, it just looks like a rectangle with one missing pixel on each corner. Now this 3px->5px is not a radical change, you still get approximately the same flat design, but the corners on interactive elements are now just a bit more rounded.
  * Focused button outline = selected_bg_color, to see what Enter key does. Example: When you close an unsaved document, you get a dialog asking whether you want to Close without Saving, or Cancel, or Save. One of those buttons is already focused, so you can just press Enter key to make that choice, or press Tab to move the focus to another button, and then press Enter. But before it was hard to routinely press the Enter key without having some doubts, because the focus was very subtly indicated with a thin dashed grey line. This line is now solid and colored, so you know what you are doing when you press that Enter key!
  * Messages colors are updated. There are various preset colors for various messages. There is the links colors, and that green color on that "Save" button, or that red color on the logout button, etc. They all got updated.  

### Express yourself
The official Mint-Y colors are currently being re-edited. Some people are saying these accent colors should be subdued. This is an understandable argument, as explained there: [https://uxmovement.com/content/why-you-should-avoid-bright-saturated-background-colors/](https://uxmovement.com/content/why-you-should-avoid-bright-saturated-background-colors/). But take this with a pinch of salt: there is a difference between a page background and the background color of a focused element. You and i, we don’t want pale greys and beiges vaguely tinted with colors. WE WANT LIVING, VIBRANT COLORS! Well, okay, it should not always be set at 100% saturation and brightness, but still... The official Mint-Y theme team seems to be going back towards the old Mint-X washed out colors. Or halfway between old and new, trying to please everyone... They claim that these old colors are more readable and easier on the eyes. I say these old colors are washed out, sad looking, and boring. My Mint-Yz theme was created as a reaction to these old colors.

Express your preferences for the official Mint-Y, before it is too late: [https://github.com/linuxmint/mint-y-colors/issues](https://github.com/linuxmint/mint-y-colors/issues). You can also [Do-It-Yourself](https://github.com/SebastJava/mint-yz-theme/blob/colors/Do-It-Yourself.md).

### Credits
This theme is based on the Mint-Y theme, which is based on the Arc theme:

  * [https://github.com/linuxmint/mint-themes](https://github.com/linuxmint/mint-themes)
  * [https://github.com/horst3180/arc-theme](https://github.com/horst3180/arc-theme)


debian/changelog
mint-yz-theme (2.4) UNRELEASED; urgency=medium

  * Subtle updates on most colors.
  * Subtle darkening on most of the dark themes.
  * Pulled small updates from the upstream source.

 -- Sébastien Bouchard <sebastjava@hotmail.ca>  Wed, 02 Dec 2020 21:37:30 -0500 

mint-yz-theme (2.3) UNRELEASED; urgency=medium

  * Subtle updates on 7 colors.

 -- Sébastien Bouchard <sebastjava@hotmail.ca>  Fri, 18 Sep 2020 13:02:00 -0400 

mint-yz-theme (2.2) UNRELEASED; urgency=medium

  * Subtle color updates on Orange, Red, Pink, and dark = light.

 -- Sébastien Bouchard <sebastjava@hotmail.ca>  Mon, 14 Sep 2020 12:52:00 -0400 

mint-yz-theme (2.1) UNRELEASED; urgency=medium

  * ALL focused buttons get a colored outline.
  * OR a white outline for the colored buttons.
  * Subtle color updates. Mint, MintContrast, Red, Pink.

 -- Sébastien Bouchard <sebastjava@hotmail.ca>  Thu, 10 Sep 2020 16:36:00 -0400 

mint-yz-theme (2.0.2) UNRELEASED; urgency=medium

  * Button focus is now indicated by a solid colored outline.

 -- Sébastien Bouchard <sebastjava@hotmail.ca>  Sun, 06 Sep 2020 19:22:00 -0400  

mint-yz-theme (2.0.1) UNRELEASED; urgency=medium

  * FIXED button focus having same color as selected.

 -- Sébastien Bouchard <sebastjava@hotmail.ca>  Sun, 06 Sep 2020 16:34:00 -0400 

mint-yz-theme (2.0) UNRELEASED; urgency=medium

  * Dark accent colors are not much darker, 10% darker or customized.
  * Color updates. Mint, MintSparkling, BlueDeep.
  * Buttons and entries rounded corners are a bit more rounded, 3px->5px.
  * Focused buttons background=selected_bg_color, to see what Enter key does.
  * Etcetera.

 -- Sébastien Bouchard <sebastjava@hotmail.ca>  Fri, 04 Sep 2020 19:56:00 -0400 

mint-yz-theme (1.6) UNRELEASED; urgency=medium

  * PurpleDeep hue 275->270 and Purple saturation 50->67.

 -- Sébastien Bouchard <sebastjava@hotmail.ca>  Sat, 22 Aug 2020 12:51:55 -0400

mint-yz-theme (1.5) UNRELEASED; urgency=medium

  * Added PurpleDeep color.

 -- Sébastien Bouchard <sebastjava@hotmail.ca>  Fri, 21 Aug 2020 15:29:02 -0400 

mint-yz-theme (1.4) UNRELEASED; urgency=medium

  * MintFresh color update.

 -- Sébastien Bouchard <sebastjava@hotmail.ca>  Fri, 07 Aug 2020 20:24:00 -0400

mint-yz-theme (1.3) UNRELEASED; urgency=medium

  * Big color updates.

 -- Sébastien Bouchard <sebastjava@hotmail.ca>  Sun, 26 Jul 2020 20:47:00 -0400

mint-yz-theme (1.2) UNRELEASED; urgency=medium

  * Aqua color update.

 -- Sébastien Bouchard <sebastjava@hotmail.ca>  Sat, 18 Jul 2020 02:28:00 -0400

mint-yz-theme (1.1) UNRELEASED; urgency=medium

  * The color of the window close button is now defined with hex value.

 -- Sébastien Bouchard <sebastjava@hotmail.ca>  Sun, 12 Jul 2020 10:34:00 -0400

mint-yz-theme (1.0) UNRELEASED; urgency=medium

  * Initial release.

 -- Sébastien Bouchard <sebastjava@hotmail.ca>  Mon, 06 Jul 2020 11:33:00 -0400


generate-themes.py
#!/usr/bin/python3
import os
import sys

from constants import Y_HEX_ACCENT1, Y_HEX_ACCENT2, Y_HEX_ACCENT3, Y_HEX_ACCENT4
from constants import y_hex_colors1, y_hex_colors2, y_hex_colors3, y_hex_colors4

def change_value (key, value, file):
    if value is not None:
        command = "sed -i '/%(key)s=/c\%(key)s=%(value)s' %(file)s" % {'key':key, 'value':value, 'file':file}
    else:
        command = "sed -i '/%(key)s=/d' %(file)s" % {'key':key, 'file':file}
    os.system(command)

# DELETED: def x_colorize_directory (path, variation):

def y_colorize_directory (path, variation):
    for accent in Y_HEX_ACCENT1:
        os.system("find %s -name '*.*' -type f -exec sed -i 's/%s/%s/gI' {}  \\;" % (path, accent, y_hex_colors1[variation]))
    for accent in Y_HEX_ACCENT2:
        os.system("find %s -name '*.*' -type f -exec sed -i 's/%s/%s/gI' {}  \\;" % (path, accent, y_hex_colors2[variation]))
    for accent in Y_HEX_ACCENT3:
        os.system("find %s -name '*.*' -type f -exec sed -i 's/%s/%s/gI' {}  \\;" % (path, accent, y_hex_colors3[variation]))
    for accent in Y_HEX_ACCENT4:
        os.system("find %s -name '*.*' -type f -exec sed -i 's/%s/%s/gI' {}  \\;" % (path, accent, y_hex_colors4[variation]))

if os.path.exists("usr"):
    os.system("rm -rf usr/")

os.system("mkdir -p usr/share/themes")

# DELETED: MINT-X GENERATE

curdir = os.getcwd()

# MINT-Y GENERATE
os.chdir("src/Mint-Y")
os.system("./build-themes.py")
os.chdir(curdir)

# Mint-Y color variations
for color in y_hex_colors1.keys():
    for variant in ["-Base", "-Darkest", "-Darker"]:
        original_name = "Mint-Yz%s" % variant
        path = os.path.join("src/Mint-Y/variations/%s" % color)
        if os.path.isdir(path):
            print("Derivating %s-%s" % (original_name, color))

            # Copy theme
            theme = "usr/share/themes/%s-%s" % (original_name, color)
            theme_index = os.path.join(theme, "index.theme")
            os.system("cp -R usr/share/themes/%s %s" % (original_name, theme))

            # Theme name
            for key in ["Name", "GtkTheme"]:
                change_value(key, "%s-%s" % (original_name, color), theme_index)

            for key in ["IconTheme"]:
                change_value(key, "Mint-Y-%s" % color, theme_index)

            for key in ["MetacityTheme"]:
                metacity_variant = original_name.replace("Darker", "Darkest")
                change_value(key, "%s-%s" % (metacity_variant, color), theme_index)

            # Regenerate GTK3 sass
            os.system("cp -R src/Mint-Y/gtk-3.0/sass %s/gtk-3.0/" % theme)
            y_colorize_directory("%s/gtk-3.0/sass" % theme, color)
            os.chdir("%s/gtk-3.0" % theme)
            if (variant == "-Darkest"):
                os.system("cp sass/gtk-dark.scss sass/gtk.scss")
                os.system("sassc ./sass/gtk.scss gtk.css")
            elif (variant == "-Darker"):
                os.system("cp sass/gtk-darker.scss sass/gtk.scss")
                os.system("sassc ./sass/gtk.scss gtk.css")
                os.system("sassc ./sass/gtk-dark.scss gtk-dark.css")
            else:
                os.system("rm sass/gtk-dark.scss sass/gtk-darker.scss")
                os.system("sassc ./sass/gtk.scss gtk.css")

            os.system("rm -rf sass .sass-cache")
            os.chdir(curdir)

            # Regenerate Cinnamon sass
            if (variant != "-Darker"):
                # Darker variants have no cinnamon style
                os.system("cp -R src/Mint-Y/cinnamon/sass %s/cinnamon/" % theme)
                y_colorize_directory("%s/cinnamon/sass" % theme, color)
                os.chdir("%s/cinnamon" % theme)
                if (variant == "-Darkest"):
                    os.system("cp sass/cinnamon-dark.scss sass/cinnamon.scss")
                os.system("sassc ./sass/cinnamon.scss cinnamon.css")
                os.system("rm -rf sass .sass-cache")
                os.chdir(curdir)

            # Accent color
            files = []
            files.append(os.path.join(theme, "gtk-2.0", "gtkrc"))
            files.append(os.path.join(theme, "gtk-2.0", "main.rc"))
            files.append(os.path.join(theme, "gtk-2.0", "panel.rc"))
            files.append(os.path.join(theme, "gtk-2.0", "apps.rc"))
            files.append(os.path.join(theme, "gtk-2.0", "menubar-toolbar", "gtkrc"))
            files.append(os.path.join(theme, "metacity-1", "metacity-theme-2.xml"))
            files.append(os.path.join(theme, "metacity-1", "metacity-theme-3.xml"))
            for file in files:
                if os.path.exists(file):
                    for accent in Y_HEX_ACCENT1:
                        os.system("sed -i s'/%(accent)s/%(color_accent)s/gI' %(file)s" % {'accent': accent, 'color_accent': y_hex_colors1[color], 'file': file})
                    for accent in Y_HEX_ACCENT2:
                        os.system("sed -i s'/%(accent)s/%(color_accent)s/gI' %(file)s" % {'accent': accent, 'color_accent': y_hex_colors2[color], 'file': file})
                    for accent in Y_HEX_ACCENT3:
                        os.system("sed -i s'/%(accent)s/%(color_accent)s/gI' %(file)s" % {'accent': accent, 'color_accent': y_hex_colors3[color], 'file': file})
                    for accent in Y_HEX_ACCENT4:
                        os.system("sed -i s'/%(accent)s/%(color_accent)s/gI' %(file)s" % {'accent': accent, 'color_accent': y_hex_colors4[color], 'file': file})

            directories = []
            directories.append(os.path.join(theme, "cinnamon/common-assets"))
            directories.append(os.path.join(theme, "cinnamon/light-assets"))
            directories.append(os.path.join(theme, "cinnamon/dark-assets"))
            for directory in directories:
                if os.path.exists(directory):
                    y_colorize_directory(directory, color)

            # Assets
            directories = []
            directories.append(os.path.join(path, "cinnamon"))
            directories.append(os.path.join(path, "gtk-2.0"))
            directories.append(os.path.join(path, "gtk-3.0"))
            directories.append(os.path.join(path, "xfwm4-src"))
            for directory in directories:
                if not os.path.exists(directory):
                    print("\nThere are missing directories in %s..." % path)
                    print("Please run './update-variations.py All' or './update-variations.py COLOR' before this './generate-themes.py'...")
                    sys.exit(1)

            os.system("rm -rf %s/cinnamon/thumbnail.png" % theme)
            os.system("rm -rf %s/gtk-2.0/assets" % theme)
            os.system("rm -rf %s/gtk-2.0/menubar-toolbar/*.png" % theme)
            os.system("rm -rf %s/gtk-3.0/assets" % theme)
            os.system("rm -rf %s/gtk-3.0/thumbnail.png" % theme)
            os.system("rm -rf %s/xfwm4" % theme)

            if variant == "-Darkest":
                os.system("cp -R %s/cinnamon/mint-y-dark-thumbnail.png %s/cinnamon/thumbnail.png" % (path, theme))
                os.system("cp -R %s/gtk-2.0/assets-dark %s/gtk-2.0/assets" % (path, theme))
                os.system("cp -R %s/gtk-3.0/thumbnail-dark.png %s/gtk-3.0/thumbnail.png" % (path, theme))
                os.system("cp -R %s/xfwm4-src/xfwm4-dark %s/xfwm4" % (path, theme))
            else:
                if variant == "-Base":
                    os.system("cp -R %s/cinnamon/mint-y-thumbnail.png %s/cinnamon/thumbnail.png" % (path, theme))
                    os.system("cp -R %s/xfwm4-src/xfwm4 %s/xfwm4" % (path, theme))
                os.system("cp -R %s/gtk-2.0/assets %s/gtk-2.0/assets" % (path, theme))
                os.system("cp -R %s/gtk-3.0/thumbnail.png %s/gtk-3.0/thumbnail.png" % (path, theme))
            os.system("cp -R %s/gtk-2.0/menubar-toolbar/*.png %s/gtk-2.0/menubar-toolbar" % (path, theme))
            os.system("cp -R %s/gtk-3.0/assets %s/gtk-3.0/assets" % (path, theme))

print ("\nDone. Are there any missing colors?")
print ("If so then run './update-variations.py All' and then './generate-themes.py' again.")


update-variations.py
#!/usr/bin/python3
import os
import sys

from constants import Y_HEX_ACCENT1, Y_HEX_ACCENT2, Y_HEX_ACCENT3, Y_HEX_ACCENT4
from constants import y_hex_colors1, y_hex_colors2, y_hex_colors3, y_hex_colors4

def change_value (key, value, file):
    if value is not None:
        command = "sed -i '/%(key)s=/c\%(key)s=%(value)s' %(file)s" % {'key':key, 'value':value, 'file':file}
    else:
        command = "sed -i '/%(key)s=/d' %(file)s" % {'key':key, 'file':file}
    os.system(command)

def usage ():
    print ("Usage: update-variations.py COLOR")
    print ("COLOR can be 'All' or one of these:")
    for color in y_hex_colors1.keys():
        print ("    %s" % color)
    sys.exit(1)

def update_color (color):
    variation = "src/Mint-Y/variations/%s" % color
    print("\n\n\nUpdating %s" % variation)
    os.system("rm -rf %s" % variation)
    os.system("mkdir -p %s/cinnamon" % variation)
    os.system("mkdir -p %s/gtk-2.0" % variation)
    os.system("mkdir -p %s/gtk-3.0" % variation)
    os.system("mkdir -p %s/xfwm4-src" % variation)

    # Copy assets files
    assets = []
    assets.append("cinnamon/mint-y-thumbnails-src.svg")
    assets.append("gtk-2.0/assets.svg")
    assets.append("gtk-2.0/assets-dark.svg")
    assets.append("gtk-3.0/assets.svg")
    assets.append("xfwm4-src/assets.svg")
    assets.append("xfwm4-src/assets-dark.svg")

    files = []
    files.append("cinnamon/render-cin-thumbnails.sh")
    files.append("gtk-2.0/assets") # copy this to delete assets/* later!?
    files.append("gtk-2.0/assets-dark") # copy this to delete assets-dark/* later!?
    files.append("gtk-2.0/assets.txt")
    files.append("gtk-2.0/menubar-toolbar")
    files.append("gtk-2.0/render-assets.sh")
    files.append("gtk-2.0/render-dark-assets.sh")
    files.append("gtk-3.0/assets") # copy this to delete assets/* later!?
    files.append("gtk-3.0/assets.txt")
    files.append("gtk-3.0/render-assets.sh")
    files.append("xfwm4-src/xfwm4")
    files.append("xfwm4-src/xfwm4-dark")
    files.append("xfwm4-src/assets.txt")
    files.append("xfwm4-src/render-assets.sh")

    for file in files:
        os.system("cp -R src/Mint-Y/%s %s/%s" % (file, variation, file))
    for asset in assets:
        os.system("cp -R src/Mint-Y/%s %s/%s" % (asset, variation, asset))

    # Update assets svg
    for asset in assets:
        asset_path = "%s/%s" % (variation, asset)
        for accent in Y_HEX_ACCENT1:
            os.system("sed -i s'/%(accent)s/%(color_accent)s/gI' %(file)s" % {'accent': accent, 'color_accent': y_hex_colors1[color], 'file': asset_path})
        for accent in Y_HEX_ACCENT2:
            os.system("sed -i s'/%(accent)s/%(color_accent)s/gI' %(file)s" % {'accent': accent, 'color_accent': y_hex_colors2[color], 'file': asset_path})
        for accent in Y_HEX_ACCENT3:
            os.system("sed -i s'/%(accent)s/%(color_accent)s/gI' %(file)s" % {'accent': accent, 'color_accent': y_hex_colors3[color], 'file': asset_path})
        for accent in Y_HEX_ACCENT4:
            os.system("sed -i s'/%(accent)s/%(color_accent)s/gI' %(file)s" % {'accent': accent, 'color_accent': y_hex_colors4[color], 'file': asset_path})

    # Render assets
    os.chdir(variation)
    os.chdir("cinnamon")
    os.system("./render-cin-thumbnails.sh")
    os.chdir("../gtk-2.0/")
    os.system("rm -rf assets/*")
    os.system("rm -rf assets-dark/*")
    os.system("./render-assets.sh")
    os.system("./render-dark-assets.sh")
    os.chdir("../gtk-3.0/")
    os.system("rm -rf assets/*")
    os.system("./render-assets.sh")
    os.chdir("../xfwm4-src/xfwm4/")
    os.system("rm -rf close-active.xpm close-prelight.xpm close-pressed.xpm")
    os.chdir("../xfwm4-dark/")
    os.system("rm -rf close-active.xpm close-prelight.xpm close-pressed.xpm")
    os.chdir("..")
    os.system("./render-assets.sh")
    os.chdir(curdir)

if len(sys.argv) < 2:
    usage()
else:
    color_variation = sys.argv[1]
    if not color_variation in y_hex_colors1.keys() and color_variation != "All":
        usage()

# Mint-Y variations
curdir = os.getcwd()

if color_variation == "All":
    if os.path.exists("src/Mint-Y/variations"):
        os.system("rm -rf src/Mint-Y/variations/")
    for color in y_hex_colors1.keys():
        update_color(color)
else:
    update_color(color_variation)


src/Mint-Y/build-themes.py
#!/usr/bin/env python3

import os

VARIATIONS = ["Mint-Yz-Base",
              "Mint-Yz-Darker",
              "Mint-Yz-Darkest"]

DEST = '../../usr/share/themes'

curdir = os.getcwd()

print("Updating Gtk3 assets")
os.chdir("gtk-3.0/")
os.system("sassc ./sass/gtk.scss gtk.css")
os.system("sassc ./sass/gtk-dark.scss gtk-dark.css")
os.system("sassc ./sass/gtk-darker.scss gtk-darker.css")
os.system("./render-assets.sh")
print("Gtk3 assets updated")

os.chdir(curdir)

print("Updating Gtk2 assets")
os.chdir("gtk-2.0/")
os.system("./render-assets.sh")
os.system("./render-dark-assets.sh")
print("Gtk2 assets updated")

os.chdir(curdir)

print("Updating Xfwm4 assets")
os.chdir("xfwm4-src/")
os.system("./render-assets.sh")
print("Xfwm4 assets updated")

os.chdir(curdir)

print("Updating Cinnamon assets")
os.chdir("cinnamon/")
os.system("sassc ./sass/cinnamon.scss cinnamon.css")
os.system("sassc ./sass/cinnamon-dark.scss cinnamon-dark.css")
print("Cinnamon assets updated")

os.chdir(curdir)

if __name__ == '__main__':
    print("Building themes")
    for variation in VARIATIONS:
        dest_folder = os.path.join(DEST, variation)
        os.system("mkdir -p %s" % dest_folder)
        if variation == "Mint-Yz-Base":
            print("    Building Mint-Yz-Base")
            os.system("cp index.theme %s/" % dest_folder)
            # Gtk2
            version_folder = os.path.join(dest_folder, "gtk-2.0")
            os.system("mkdir -p %s" % version_folder)
            os.system("cp -R gtk-2.0/assets %s" % version_folder)
            os.system("cp -R gtk-2.0/menubar-toolbar %s" % version_folder)
            os.system("cp gtk-2.0/*.rc %s" % version_folder)
            os.system("cp gtk-2.0/gtkrc %s" % version_folder)
            # Gtk3
            version_folder = os.path.join(dest_folder, "gtk-3.0")
            os.system("mkdir -p %s" % version_folder)
            os.system("cp -R gtk-3.0/assets %s" % version_folder)
            os.system("cp gtk-3.0/gtk.css %s" % version_folder)
            os.system("cp gtk-3.0/thumbnail.png %s" % version_folder)
            # Metacity
            os.system("cp -R metacity-1 %s" % dest_folder)
            os.system("rm %s/*-dark*" % (os.path.join(dest_folder, "metacity-1")))
            # Cinnamon
            version_folder = os.path.join(dest_folder, "cinnamon")
            os.system("mkdir -p %s" % version_folder)
            os.system("cp -R cinnamon/common-assets %s" % version_folder)
            os.system("cp -R cinnamon/light-assets %s" % version_folder)
            os.system("cp cinnamon/mint-y-thumbnail.png %s" % os.path.join(version_folder, "thumbnail.png"))
            os.system("cp cinnamon/cinnamon.css %s" % version_folder)
            # XFWM
            os.system("cp -R xfwm4-src/xfwm4 %s" % dest_folder)

        elif variation == "Mint-Yz-Darker":
            print("    Building Mint-Yz-Darker")
            os.system("cp index.theme-darker %s" % os.path.join(dest_folder, "index.theme"))
            # Gtk2
            version_folder = os.path.join(dest_folder, "gtk-2.0")
            os.system("mkdir -p %s" % version_folder)
            os.system("cp -R gtk-2.0/assets %s" % version_folder)
            os.system("cp -R gtk-2.0/menubar-toolbar %s" % version_folder)
            os.system("cp gtk-2.0/*.rc %s" % version_folder)
            os.system("cp gtk-2.0/gtkrc-darker %s" % os.path.join(version_folder, "gtkrc"))
            # Gtk3
            version_folder = os.path.join(dest_folder, "gtk-3.0")
            os.system("mkdir -p %s" % version_folder)
            os.system("cp -R gtk-3.0/assets %s" % version_folder)
            os.system("cp gtk-3.0/gtk-darker.css %s" % os.path.join(version_folder, "gtk.css"))
            os.system("cp gtk-3.0/gtk-dark.css %s" % version_folder)
            os.system("cp gtk-3.0/thumbnail.png %s" % version_folder)

        elif variation == "Mint-Yz-Darkest":
            print("    Building Mint-Yz-Darkest")
            os.system("cp index.theme-dark %s" % os.path.join(dest_folder, "index.theme"))
            # Gtk2
            version_folder = os.path.join(dest_folder, "gtk-2.0")
            os.system("mkdir -p %s" % version_folder)
            os.system("cp -R gtk-2.0/assets-dark %s" % version_folder)
            os.system("rm -rf %s" % os.path.join(version_folder, "assets"))
            os.system("mv %s %s" % (os.path.join(version_folder, "assets-dark"), os.path.join(version_folder, "assets")))
            os.system("cp -R gtk-2.0/menubar-toolbar %s" % version_folder)
            os.system("cp gtk-2.0/*.rc %s" % version_folder)
            os.system("cp gtk-2.0/gtkrc-dark %s" % os.path.join(version_folder, "gtkrc"))
            # Gtk3
            version_folder = os.path.join(dest_folder, "gtk-3.0")
            os.system("mkdir -p %s" % version_folder)
            os.system("cp -R gtk-3.0/assets %s" % version_folder)
            os.system("cp gtk-3.0/gtk-dark.css %s" % os.path.join(version_folder, "gtk.css"))
            os.system("cp gtk-3.0/thumbnail-dark.png %s" % os.path.join(version_folder, "thumbnail.png"))
            # Metacity
            os.system("cp -R metacity-1 %s" % dest_folder)
            os.system("mv %s %s" % (os.path.join(dest_folder, "metacity-1", "metacity-theme-2-dark.xml"), os.path.join(dest_folder, "metacity-1", "metacity-theme-2.xml")))
            os.system("mv %s %s" % (os.path.join(dest_folder, "metacity-1", "metacity-theme-3-dark.xml"), os.path.join(dest_folder, "metacity-1", "metacity-theme-3.xml")))
            os.system("mv %s %s" % (os.path.join(dest_folder, "metacity-1", "thumbnail-dark.png"), os.path.join(dest_folder, "metacity-1", "thumbnail.png")))
            # Cinnamon
            version_folder = os.path.join(dest_folder, "cinnamon")
            os.system("mkdir -p %s" % version_folder)
            os.system("cp -R cinnamon/common-assets %s" % version_folder)
            os.system("cp -R cinnamon/dark-assets %s" % version_folder)
            os.system("cp cinnamon/mint-y-dark-thumbnail.png %s" % os.path.join(version_folder, "thumbnail.png"))
            os.system("cp cinnamon/cinnamon-dark.css %s" % os.path.join(version_folder, "cinnamon.css"))
            # XFWM
            os.system("rm -rf %s" % os.path.join(dest_folder, "xfwm4"))
            os.system("cp -R xfwm4-src/xfwm4-dark %s" % dest_folder)
            os.system("mv %s %s" % (os.path.join(dest_folder, "xfwm4-dark"), os.path.join(dest_folder, "xfwm4")))


constants.py
#!/usr/bin/python3

# MINT-Yz

# colors1 IS THE MAIN ACCENT COLOR for the light themes.
# colors2 IS ONLY FOR the dark (or darker?) variants, except some colors2 also found in a few elements in /usr/share/themes/Mint-Y-Aqua/cinnamon/common-assets/
# colors3 IS ONLY FOR ~/mint-themes/src/Mint-Y/variations/Aqua/gtk-3.0/assets.svg: titlebutton-close-hover
# colors4 IS ONLY FOR ~/mint-themes/src/Mint-Y/variations/Aqua/gtk-3.0/assets.svg: titlebutton-close-active
# At least this was true as of 2020-06-07 (YYYY-MM-DD).

Y_HEX_ACCENT1 = ["#92b372"]  # BASE
Y_HEX_ACCENT2 = ["#8fa876"]  # DARK
Y_HEX_ACCENT3 = ["#afca95"]  # PRELIGHT/HOVER
Y_HEX_ACCENT4 = ["#779559"]  # PRESSED

y_hex_colors1 = {}
y_hex_colors2 = {}
y_hex_colors3 = {}
y_hex_colors4 = {}

y_hex_colors1["MintSoda"] = "#40BF40"
y_hex_colors2["MintSoda"] = "#3DB63D"
y_hex_colors3["MintSoda"] = "#66CC66"
y_hex_colors4["MintSoda"] = "#2D862D"

y_hex_colors1["MintSoft"] = "#14B866"
y_hex_colors2["MintSoft"] = "#13AF61"
y_hex_colors3["MintSoft"] = "#43C685"
y_hex_colors4["MintSoft"] = "#0E8148"

y_hex_colors1["Grey"] = "#767676"
y_hex_colors2["Grey"] = "#767676"
y_hex_colors3["Grey"] = "#919191"
y_hex_colors4["Grey"] = "#535353"

y_hex_colors1["Red"] = "#FF1428"
y_hex_colors2["Red"] = "#F20D20"
y_hex_colors3["Red"] = "#FF4353"
y_hex_colors4["Red"] = "#B30E1C"

y_hex_colors1["Orange"] = "#FF7200"
y_hex_colors2["Orange"] = "#FF7200"
y_hex_colors3["Orange"] = "#FF8E33"
y_hex_colors4["Orange"] = "#B35000"

y_hex_colors1["Aqua"] = "#00AACC"
y_hex_colors2["Aqua"] = "#00A2C2"
y_hex_colors3["Aqua"] = "#33BBD6"
y_hex_colors4["Aqua"] = "#00778F"

y_hex_colors1["Blue"] = "#1A80E5"
y_hex_colors2["Blue"] = "#197ADA"
y_hex_colors3["Blue"] = "#4899EA"
y_hex_colors4["Blue"] = "#125AA1"

y_hex_colors1["BlueDeep"] = "#3753E2"
y_hex_colors2["BlueDeep"] = "#3753E2"
y_hex_colors3["BlueDeep"] = "#5F75E8"
y_hex_colors4["BlueDeep"] = "#273A9F"

y_hex_colors1["PurpleDeep"] = "#8033CC"
y_hex_colors2["PurpleDeep"] = "#8033CC"
y_hex_colors3["PurpleDeep"] = "#995CD6"
y_hex_colors4["PurpleDeep"] = "#5A248F"

y_hex_colors1["Purple"] = "#AA40BF"
y_hex_colors2["Purple"] = "#A23DB6"
y_hex_colors3["Purple"] = "#BB66CC"
y_hex_colors4["Purple"] = "#772D86"

y_hex_colors1["Pink"] = "#FF4DA6"
y_hex_colors2["Pink"] = "#F53D9A"
y_hex_colors3["Pink"] = "#FF71B8"
y_hex_colors4["Pink"] = "#B33675"

